src/main/java/seedu/recipe/model/cooked/UniqueRecordList.java

```diff
@@ -140,16 +144,49 @@ private boolean recordsAreUnique(List<Record> records) {
 
             List<Goal> currGoals = new ArrayList<Goal>();
             currGoals.addAll(internalList.get(i).getGoals());
-            for (Goal curr : currGoals) {
-                String goalName = curr.goalName;
-                Integer prevCount = goalMap.get(goalName);
-                goalMap.put(goalName, prevCount + 1);
+            for (Goal currGoal : currGoals) {
+                String goalName = currGoal.goalName;
+                if (goalMap.containsKey(goalName)) {
+                    Integer prevCount = goalMap.get(goalName);
+                    goalMap.put(goalName, prevCount + 1);
+                }
             }
         }
-        ObservableList<Integer> internalGoalList = FXCollections.observableArrayList(
+        internalGoalsList.addAll(
                 goalMap.get("Herbivore"),
                 goalMap.get("Bulk like the Hulk"),
                 goalMap.get("Wholesome Wholemeal"));
-        return FXCollections.unmodifiableObservableList(internalGoalList);
+        return internalUnmodifiableGoalsList;
+    }
+
+    /**
+     * Updates goals for a record.
+     * @param record
+     */
+    public void updateGoalsTally(Record record) {
+        List<Goal> currGoals = new ArrayList<Goal>();
+        currGoals.addAll(record.getGoals());
+        for (Goal currGoal: currGoals) {
+            String goalName = currGoal.goalName;
+            int currCount = 0;
+            switch(goalName) {
+
+            case "Herbivore":
+                currCount = internalGoalsList.get(0);
+                internalGoalsList.set(0, currCount + 1);
```

Comment:
I think it would be good to add that 0, 1 and 2 refers to "herbivore", "bulk like the hulk" and "wholesome wholemeal" somewhere in the javadoc